app/src/main/java/fr/lille1/univ/android/architecture/btrshop/scanner/ScannerActivity.java

```diff
@@ -0,0 +1,241 @@
+package fr.lille1.univ.android.architecture.btrshop.scanner;
+
+import android.content.Intent;
+import android.media.Ringtone;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.view.MenuItemCompat;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.ViewGroup;
+
+import fr.lille1.univ.android.architecture.btrshop.R;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.Result;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.lille1.univ.android.architecture.btrshop.scanner.domain.dialog.CameraSelectorDialogFragment;
+import fr.lille1.univ.android.architecture.btrshop.scanner.domain.dialog.FormatSelectorDialogFragment;
+import fr.lille1.univ.android.architecture.btrshop.scanner.domain.dialog.MessageDialogFragment;
+import me.dm7.barcodescanner.zxing.ZXingScannerView;
+
+public class ScannerActivity extends AppCompatActivity implements MessageDialogFragment.MessageDialogListener,
```

Comment:
That's a LOT of code for just a scan.
I think most / all of the values can be set by default (like auto focus ? of course), always back camera (i'm not going to scan with my front cam), flash probably OFF, et format.. well... all ?

Cleaning all of these you'll have a short / sexy class.